### PR TITLE
You can sell stuff on the blackmarket now.

### PR DESCRIPTION
**Pull request by Ghommie**
## About The Pull Request
The idea came up during the last blackmarket-related PR (#85066), when the original creator @TheChosenEvilOne came and said the LTSRBT was originally supposed to allow players to sell stuff on the blackmarket. I replied saying the idea has some potential, and then other github users followed, also saying it's a good idea.

So, here we are, adding another feature to the LTSRBT, to let you provide other players with potentially knock-offs and fraudulent (or genuinely honest) items on the blackmarket.

How you do it is fairly simple: left click the machine to open it, place the item inside, then right-click to open the UI. From there you can adjust the name, description and price of the market item. Finally, click on the "Place on Market" button twice, pay a 30 credits fee, and it's done. You have an item on the market, and once it's sold, 85% or the earnings (the price) will be deposited on your account.

By the by, the icon of the item you're trying to sell will also show up on the market. You should get a picture of what you're getting before falling for an obvious scam.

screenie

By the by, it's a cardboard cutout.

## Why It's Good For The Game
This fits the design of the LTSRBT and blackmarket in general. 

## Changelog

:cl:
add: You can sell items on the blackmarket with the LTSRBT now.
fix: Added some checks to prevent the swapper device and bluespace anomalies from theorically being able to send things and people to nullspace.
/:cl:

### CODE REVIEW

**Comment by Paxilmaniac** [comment]
its peak

File: tgui/packages/tgui/interfaces/LTSRBT.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+  Input,
+  NumberInput,
+  Section,
+  Stack,
+  TextArea,
+} from '../components';
+import { Window } from '../layouts';
+
+export const LTSRBT = (props) => {
+  const { act, data } = useBackend();
```
Comment (jlsnow301) [inline tgui/packages/tgui/interfaces/LTSRBT.jsx:16]: please destructure `data` after this line rather than using `data.*`

**Review by jlsnow301 — CHANGES_REQUESTED** [review]
> .jsx

**Comment by Dawnseer** [comment]
Time to get a new hobby as a cargo tech

**Comment by Ghommie** [comment]
> step 1 ian (uncontained) step 2 99999999cr step 3 ian contained (forever)

you cannot sell things containing mobs (pai, mob holders, bluespace body bags with people, soulshard), because of some complications with sending mobs to nullspace. and there's a cap of 9999 credits by the by,

**Comment by CCC23** [comment]
SELLING NUKE DISK 20 CR buy today

**Comment by deathrobotpunch** [comment]
> SELLING NUKE DISK 20 CR buy today

I think sending the nuke disk to null space resets it

**Comment by moocowswag** [comment]
> > SELLING NUKE DISK 20 CR buy today
> 
> I think sending the nuke disk to null space resets it

Can someone still buy it on black market ui? does it dupe / move the item again?

**Review by jlsnow301 — APPROVED** [review]

**Comment by ArcaneMusic** [comment]
I'm going to spend a minute to go through this for balance purposes before merge if that's alright, just clearing up some wallening stuff at the second.

**Comment by Ghommie** [comment]
> > > SELLING NUKE DISK 20 CR buy today
> > 
> > 
> > I think sending the nuke disk to null space resets it
> 
> Can someone still buy it on black market ui? does it dupe / move the item again?

Nope. If the item cannot be contained by the machine or is deleted orgoes back on the station when we try to sell it, it won't appear in the market ui at all.

**Comment by CCC23** [comment]
selling totalyl real diskie 2000 cr buy now please trust me guys its real

**Comment by shellspeed1** [comment]
Any chance this can be setup to sell multiple items for the sake of packing large amounts of drugs or meds into it?

**Comment by Dawnseer** [comment]
> Any chance this can be setup to sell multiple items for the sake of packing large amounts of drugs or meds into it?

Wouldn't you be able to do this just by selling a box full of the stuff you want to sell? Let the customers decide if they're willing to take the risk of an empty box. Or that they have to pay the price for the full package.

**Comment by billiumbillious** [comment]
is there a way to give additional pictures of the thing your selling?  imo it'd be quite convenient for selling stuff in boxes

File: code/modules/cargo/markets/market_telepad.dm [hunk 2]
```diff
@@ -48,40 +56,274 @@
...
+		return ITEM_INTERACT_BLOCKING
+
+	if(istype(tool, /obj/item/holochip))
+		var/obj/item/holochip/chip = tool
+		chip.spend(restock_cost)
+	else
+		qdel(tool)
+		if(creds_value != restock_cost)
+			var/obj/item/holochip/change = new(creds_value - restock_cost)
+			user.put_in_hands(change)
```
Comment (ArcaneMusic) [inline code/modules/cargo/markets/market_telepad.dm:206]: In testing, this seems to produce a 1 holocredit chip when trying to restock the market with no-holocredit money sources. Tested via c10000 bills on the machine as well as with smaller bills.

**Review by ArcaneMusic — CHANGES_REQUESTED** [review]
Two small things, otherwise looks good

**Review by ArcaneMusic — APPROVED** [review]

**Review by Thommi2191 — APPROVED** [review]

**Pull request merged**